packages/@n8n/api-types/src/schemas/insights.schema.ts

```diff
@@ -85,18 +85,9 @@ export const insightsByTimeDataSchemas = {
 export const insightsByTimeSchema = z.object(insightsByTimeDataSchemas).strict();
 export type InsightsByTime = z.infer<typeof insightsByTimeSchema>;
 
-export const INSIGHTS_DATE_RANGE_KEYS = [
-	'day',
-	'week',
-	'2weeks',
-	'month',
-	'quarter',
-	'6months',
-	'year',
-] as const;
 export const insightsDateRangeSchema = z
 	.object({
-		key: z.enum(INSIGHTS_DATE_RANGE_KEYS),
+		key: z.enum(['day', 'week', '2weeks', 'month', 'quarter', '6months', 'year']),
```

Comment:
Once a module's types are in the module, we can remove the duplication by getting this from `insights.constants.ts`.